src/v/cloud_storage/remote_segment.cc

```diff
@@ -454,6 +617,35 @@ ss::future<> remote_segment::hydrate() {
     });
 }
 
+ss::future<std::vector<cluster::rm_stm::tx_range>>
+remote_segment::aborted_transactions(model::offset from, model::offset to) {
+    co_await hydrate();
+    std::vector<cluster::rm_stm::tx_range> result;
+    if (!_tx_range) {
+        // We got NoSuchKey when we tried to download the
```

Comment:
maybe upload empty tx-manifest in case there's no aborted transactions? Cause otherwise we can't distinguish incomplete data from data with no aborted transactions by just looking in S3.

Reply:
if segment was uploaded but tx-manifest wasn't we won't update the archival snapshot and will retry, so it's not ambiguous